index.js

```diff
@@ -4,141 +4,145 @@ const BridgeMethod = require("./BridgeMethod");
 const BridgeEvent = require("./BridgeEvent");
 const utils = require("./utils");
 
-const getBridgeTransactionByTxHash = async (web3Client, transactionHash) => {
-    utils.verifyHashOrBlockNumber(transactionHash);
+class BridgeTransactionParser {
 
-    let transaction;
-    const txReceipt = await web3Client.eth.getTransactionReceipt(transactionHash);
-    
-    if (txReceipt?.to === Bridge.address) {
-        const bridge = Bridge.build(web3Client);
-        const tx = await web3Client.eth.getTransaction(txReceipt.transactionHash);
-        transaction = await createBridgeTx(web3Client, bridge, tx, txReceipt);
-    }
-    
-    return transaction;
-}
-
-const getBridgeTransactionsInThisBlock = async (web3Client, blockHashOrBlockNumber) => {
-    utils.verifyHashOrBlockNumber(blockHashOrBlockNumber);
-
-    const block = await web3Client.eth.getBlock(blockHashOrBlockNumber);
-    if (!block) {
-        throw new Error(`Block ${blockHashOrBlockNumber} not found`);
-    }
-
-    const bridgeTxs = [];
-    for (let txHash of block.transactions) {
-        let transaction = await getBridgeTransactionByTxHash(web3Client, txHash);
-        if (transaction) {
-            bridgeTxs.push(transaction);
+    constructor(web3Client) {
+        if (!web3Client) {
+            throw new Error(`web3Client is required`);
         }
+        this.web3Client = web3Client;
```

Comment:
Shouldn't the web3Client be declared somewhere?

Reply:
No, no need to. This is JS. This property gets assigned dynamically to the `this` object at this point. With TS we would declare it.